File: Example/ColorPicker/ViewController.m
//
//  ViewController.m
//  ColorPicker
//
//  Created by Fabian Canas on 12/6/14.
//  Copyright (c) 2014 Fabián Cañas. All rights reserved.
//

#import "ViewController.h"

#import <iOS-Color-Picker/FCColorPickerViewController.h>

@interface ViewController () <FCColorPickerViewControllerDelegate>
@property (nonatomic, copy) UIColor *color;
@end

@implementation ViewController

- (IBAction)pickColor:(id)sender {
    FCColorPickerViewController *colorPicker = [FCColorPickerViewController colorPickerWithColor:self.color
                                                                                        delegate:self];
    colorPicker.tintColor = [UIColor whiteColor];
    [colorPicker setModalPresentationStyle:UIModalPresentationFormSheet];
    [self presentViewController:colorPicker
                       animated:YES
                     completion:nil];
}

- (void)colorPickerViewController:(FCColorPickerViewController *)colorPicker
                   didSelectColor:(UIColor *)color
{
    self.color = color;
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)colorPickerViewControllerDidCancel:(FCColorPickerViewController *)colorPicker
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)setColor:(UIColor *)color
{
    _color = [color copy];
    [self.view setBackgroundColor:_color];
}

@end

File: Example/SwiftyPicker/ViewController.swift
//
//  ViewController.swift
//  SwiftyPicker
//
//  Created by Fabian Canas on 9/14/15.
//  Copyright © 2015 Fabián Cañas. All rights reserved.
//

import UIKit
import iOS_Color_Picker

class ViewController: UIViewController, FCColorPickerViewControllerDelegate {

    var color = UIColor.blue {
        didSet {
            self.view?.backgroundColor = color
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func pickColor(_ sender :AnyObject) {
        let colorPicker = FCColorPickerViewController()
        colorPicker.color = color
        colorPicker.delegate = self
        present(colorPicker, animated: true, completion: nil)
    }
    
    func colorPickerViewController(_ colorPicker: FCColorPickerViewController, didSelect color: UIColor) {
        self.color = color
        dismiss(animated: true, completion: nil)
    }

    func colorPickerViewControllerDidCancel(_ colorPicker: FCColorPickerViewController) {
        dismiss(animated: true, completion: nil)
    }
    
}

